File: 819 - Most Common Word.java
import java.util.*;

class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        paragraph += ".";

        Set<String> banset = new HashSet<>();
        for (String s : banned) {
            banset.add(s);
        }

        Map<String, Integer> m = new HashMap<>();

        StringBuilder sb = new StringBuilder();
        for (Character c : paragraph.toCharArray()) {
            if (Character.isLetter(c)) {
                sb.append(Character.toLowerCase(c));
            } else if (sb.length() > 0) {
                String word = sb.toString();
                if (!banset.contains(word)) {
                    m.put(word, m.getOrDefault(word, 0) + 1);
                }
                sb = new StringBuilder();
            }
        }
        int max = 0;
        String res = new String();
        for (Map.Entry<String, Integer> e : m.entrySet()) {
            if (max < e.getValue()) {
                res = e.getKey();
                max = e.getValue();
            }
        }
        return res;
    }
}

File: 819 - Most Common Word.js
/**
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function(paragraph, banned) {
    let s = new Set(banned);
    let m = new Map();
    let temp = '';
    let patt = /[a-zA-Z]/;
    paragraph = paragraph.toLowerCase();
    paragraph += '.';
    for (let c of paragraph) {
        if (patt.test(c)) {
            temp += c;
        } else if (temp.length > 0) {
            if (!s.has(temp)) {
                if (!m.has(temp)) {
                    m.set(temp, 0);
                }
                m.set(temp, m.get(temp) + 1);
            }
            temp = '';
        }
    }
    let res = '';
    let max = 0;
    for (let [word, count] of m.entries()) {
        
        if (max < count) {
            max = count;
            res = word;
        }
    }
    return res;
};